### PR TITLE
Feature/messages ui with job system

### PR DESCRIPTION
**Pull request by maxwellpark**
Integrating the messages and jobs systems. 

If a message contains a job offer, a button will appear in the message detail view. Click that button to accept the job. The job will then show up in the jobs panel's available jobs container. Drag it into the schedule container to schedule the job.

### CODE REVIEW

File: Assets/Scripts/CharacterCustomisation/ColourSliders.cs [hunk 1]
```diff
@@ -21,12 +21,12 @@ public class ColourSliders : MonoBehaviour
 
     private void Start()
     {
-        InitialiseSliderContainer();
-        InitialiseImageObject();
+        //InitialiseSliderContainer();
+        //InitialiseImageObject();
```
Comment (ambid17) [inline Assets/Scripts/CharacterCustomisation/ColourSliders.cs]: bad

File: Assets/Scripts/JobSystem/DragNDrop.cs [hunk 2]
```diff
@@ -49,10 +53,16 @@ public void OnEndDrag(PointerEventData eventData)
...
-        if (gameObject.transform.parent != scheduleContainer.transform)
+        // Return to starting position if not dropped in a slot  
+        if (!jobsUI.IsInsideSlot(gameObject))
         {
             gameObject.transform.parent = availableJobsContainer.transform;
+
+            // If the job was previously scheduled, update the schedule accordingly 
+            if (job.isScheduled)
+            {
+                onUnscheduleJob?.Invoke(job);
```
Comment (ambid17) [inline Assets/Scripts/JobSystem/DragNDrop.cs:62]: the name `onUnscheduleJob` seems unintuitive here

File: Assets/Scripts/JobSystem/JobsUI.cs [hunk 3]
```diff
@@ -61,8 +64,8 @@ private GameObject InitialiseScheduleContainer()
         Image scheduleImage = scheduleObject.AddComponent<Image>();
         scheduleImage.sprite = containerBackgroundSprite;
 
-        scheduleObject.AddComponent<CanvasGroup>();
-        scheduleObject.AddComponent<Schedule>(); 
+        scheduleObject.AddComponent<CanvasGroup>().blocksRaycasts = true; 
+        //scheduleObject.AddComponent<ScheduleUI>();
```
Comment (ambid17) [inline Assets/Scripts/JobSystem/JobsUI.cs]: bad

File: Assets/Scripts/JobSystem/JobsUI.cs [hunk 4]
```diff
@@ -61,8 +64,8 @@ private GameObject InitialiseScheduleContainer()
         Image scheduleImage = scheduleObject.AddComponent<Image>();
         scheduleImage.sprite = containerBackgroundSprite;
 
-        scheduleObject.AddComponent<CanvasGroup>();
-        scheduleObject.AddComponent<Schedule>(); 
+        scheduleObject.AddComponent<CanvasGroup>().blocksRaycasts = true;
```
Comment (ambid17) [inline Assets/Scripts/JobSystem/JobsUI.cs:73]: this is a hack if ive ever seen one 😂

File: Assets/Scripts/JobSystem/JobsUI.cs [hunk 5]
```diff
@@ -6,21 +6,20 @@
...
+    public GameObject scheduleSlotPrefab; 
     public JobsContainer jobsContainer;
-    public GameObject jobPrefab; 
+    public Schedule schedule; 
+
     private GameObject availableJobsContainer;
     private GameObject scheduleContainer;
 
     private Sprite containerBackgroundSprite;
     private Sprite scheduleSlotSprite;
```
Comment (ambid17) [inline Assets/Scripts/JobSystem/JobsUI.cs:19]: unused

File: Assets/Scripts/UI/MainTabbedUIManager.cs [hunk 6]
```diff
@@ -7,19 +7,21 @@ public class MainTabbedUIManager : MonoBehaviour
...
     public GameObject messagesPanel;
+    public GameObject jobsPanel; 
     public GameObject analyticsPanel;
     public GameObject crewPanel;
     public GameObject upgradesPanel;
 
     public enum Tab
     {
-        missions, messages, analytics, crew, upgrades
+        missions, messages, jobs, analytics, crew, upgrades
```
Comment (ambid17) [inline Assets/Scripts/UI/MainTabbedUIManager.cs]: typically enum values are capitalized

**Review by ambid17 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/JobSystem/JobsUI.cs (hunk 4 above)
Comment (maxwellpark) [reply Assets/Scripts/JobSystem/JobsUI.cs:73]: lol it's not exactly ideal, but will it suffice for now?

File: Assets/Scripts/JobSystem/DragNDrop.cs (hunk 2 above)
Comment (maxwellpark) [reply Assets/Scripts/JobSystem/DragNDrop.cs:62]: Well this logic fires if the player drags the job out of the schedule. In this scenario, it was already scheduled (already inside the schedule container), then it becomes unscheduled (goes back to the available jobs container). I think it's okay personally. But if you have a better name, I'm all ears. 

The comment is a tad misleading though. Will change that.

File: Assets/Scripts/JobSystem/JobsUI.cs (hunk 3 above)
Comment (maxwellpark) [reply Assets/Scripts/JobSystem/JobsUI.cs]: Just the commented out line right?

File: Assets/Scripts/JobSystem/JobsUI.cs (hunk 5 above)
Comment (maxwellpark) [reply Assets/Scripts/JobSystem/JobsUI.cs:19]: It is used to initialise the schedule slots: `for (int i = 1; i <= schedule.numberOfDays; i++)` 

But nowhere else it seems. I could move numberOfDays to a static member of a constants class or have it as a member of JobsUI. At the moment, it is stored in the schedule scriptableobject. 

What should I do with it?

File: Assets/Scripts/JobSystem/JobsUI.cs (hunk 4 above)
Comment (ambid17) [reply Assets/Scripts/JobSystem/JobsUI.cs:73]: yeah its fine, it was just funny

File: Assets/Scripts/JobSystem/JobsUI.cs (hunk 5 above)
Comment (ambid17) [reply Assets/Scripts/JobSystem/JobsUI.cs:19]: oh, i did a `find all references` and it didn't find any usages, must be a VS bug

**Review by ambid17 — APPROVED** [review]

**Pull request merged**